test/helpers/votingTestCases.js

```diff
@@ -0,0 +1,188 @@
+import { expect } from "chai";
+import { increaseTime } from "../helpers/evmCommands";
+import { createSignedVote } from "../helpers/relayedVoteSigning";
+import { prettyPrintGasCost } from "../helpers/gasUtils";
+
+export function testVotingWithSnapshots(getToken, getVotingContract, owner, owner2, relayer) {
+  let token;
+  let votingContract;
+
+  beforeEach(() => {
+    votingContract = getVotingContract();
+    token = getToken();
+  });
+
+  describe("Opening Proposals:", () => {
+    it("should allow to open a proposal", async () => {
+      const addProposalLog = await votingContract.addProposal();
+      await expect(addProposalLog.logs.find(log => log.event === "LogNewProposal")).to.not.be.empty;
```

Comment:
actually we have a set of nice helper functions for testing logs. and we test every field in the log of course.